examples/typescript/solana/account.transfer.ts

```diff
@@ -0,0 +1,74 @@
+// Usage: pnpm tsx solana/account.transfer.ts
+
+import { CdpClient } from "@coinbase/cdp-sdk";
+import "dotenv/config";
+
+const cdp = new CdpClient();
+
+const sender = await cdp.solana.getOrCreateAccount({
+  name: "Sender",
+});
+
+const amount = "0.0001";
+
+await faucetIfNeeded(sender.address, amount);
+
+const transfer = await sender.transfer({
+  to: "3KzDtddx4i53FBkvCzuDmRbaMozTZoJBb1TToWhz3JfE",
+  amount,
+  token: "sol",
+  network: "devnet",
+});
+
+console.log(
+  `Sent transaction with signature: ${transfer.signature}. Waiting for confirmation...`
+);
+
+const { signature } = await sender.waitForTransactionConfirmation({
+  signature: transfer.signature,
+  network: "devnet",
+});
+
+console.log(
+  `Transaction confirmed: Link: https://explorer.solana.com/tx/${signature}?cluster=devnet`
+);
+
+async function faucetIfNeeded(address: string, amount: string) {
+  function sleep(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  const { Connection, PublicKey, LAMPORTS_PER_SOL } = await import(
+    "@solana/web3.js"
+  );
+  const connection = new Connection("https://api.devnet.solana.com");
+
+  const lamportsToSend = Number(amount) * LAMPORTS_PER_SOL;
+
+  // Wait until the address has balance
+  let balance = 0;
+
+  if (balance < lamportsToSend) {
+    await sender.requestFaucet({
+      token: "sol",
+    });
+  }
+
+  let attempts = 0;
+  const maxAttempts = 30;
+
+  while (balance === 0 && attempts < maxAttempts) {
```

Comment:
if `lamportsToSend` is 0, this code wouldn't request funds but would still spend 30 seconds trying to fetch a positive balance.